core/src/main/java/org/dogtagpki/tomcat/JSSNioEndpoint.java

```diff
@@ -0,0 +1,77 @@
+package org.dogtagpki.tomcat;
```

Comment:
The methods from this and the following class come from tomcat. Should we put a note to indicate the origin?

Reply:
I'm not sure about the copyright/licensing issue, but is it possible to make some changes in Tomcat so that the code that we need to copy is minimized? For now we just need to make the changes in downstream Tomcat, but later we can try to merge it upstream too.

Reply:
Regardless of copyright/licensing issue, I'd suggest to document the origin of the code if it resembles an existing code somewhere else.

Reply:
@fmarco76 I'm not familiar with Tomcat enough to make any assessment, but in general, you are right to be concerned about copying the code.  I'm with @edewata there.  Is there a way to talk to the Tomcat team to make an adjustment so that the class will become extensible in some way? Also, if you are allowed to copy the code, I'd suggest that you add a comment in there detailing where the code came from, and if you had made changes to the copied code, create a block (use comments to state begin and end of changed code) so the readers know exactly what has been changed.

Reply:
Added some comments about the origin of the code and what has been modified